specutils/io/default_loaders/dc_common.py

```diff
@@ -352,9 +357,7 @@ def add_single_spectra_to_map(
         spectrum.uncertainty = UNCERTAINTY_MAP[purpose](aligned_flux)
         spectrum.meta["uncertainty_header"] = header
 
-    # We never actually want to return something, this just flags it to pylint
-    # that we know we're breaking out of the function when skip is selected
-    return None
+    return spectrum
```

Comment:
Why the change here? It looks like this function acts on an existing `spectra_map` object and thus, as the comment says, shouldn't return anything.

Reply:
The original idea (based on what spectra I had access to at the time) was that files were either a single extension (with data represented as a 2d array, which is what most of the GAMA spectra were) or were across multiple extensions. We've got (legacy, so cannot change the format the files were written in) files that instead require more manipulation (e.g. the WCS was only in the first extension), so we want to be able to keep a reference around to the 1d spectrum independent of the mapping.

There are at least 3 surveys where we're going to use this (MGC, SAMI, S7), so rather than hacking this into each of their loaders, I figured I'd do the change in the place where it needs to be.

Reply:
Gotcha, seems reasonable enough. Any chance you could add a test for the new fallback header code?